becquerel/core/energycal.py

```diff
@@ -46,28 +46,43 @@ def __init__(self):
         # initialize fit constraints?
 
     @classmethod
-    def from_points(cls, chlist, kevlist):
+    def from_points(cls, chlist, kevlist, include_origin=False):
         """Construct EnergyCal from calibration points.
 
         Args:
           chlist: list/tuple/array of the channel values of calibration points
           kevlist: list/tuple/array of the corresponding energy values [keV]
+          include_origin: Default is False, if set to True will add a point
+                          at zero into the fit.
 
         Raises:
           BadInput: for bad chlist and/or kevlist.
         """
 
         if chlist is None or kevlist is None:
             raise BadInput('Channel list and energy list are required')
-        elif (not isinstance(chlist, VECTOR_TYPES) or
-                not isinstance(kevlist, VECTOR_TYPES)):
+
+        cond1 = isinstance(chlist, VECTOR_TYPES)
+        cond2 = isinstance(kevlist, VECTOR_TYPES)
+
+        if include_origin:
+            if cond1 and cond2:
+                chlist = np.append(0, chlist)
+                kevlist = np.append(0, kevlist)
+            elif not cond1 and not cond2:
+                chlist = [0, chlist]
+                kevlist = [0, kevlist]
```

Comment:
@cosama, I think this will allow scalar `chlist` and `kevlist` which is confusing because that will `raise` a `BadInput` exception below if `inlucude_origin=False`. Did you have a specific case in mind when including this `elif`? Otherwise I think this logic could be simplified to only allow `VECTOR_TYPES` input